pkg/server/ingest.go

```diff
@@ -100,6 +101,12 @@ func ingestParamsFromRequest(r *http.Request) *ingestParams {
 }
 
 func (ctrl *Controller) ingestHandler(w http.ResponseWriter, r *http.Request) {
+	if resource_usages.IsRunningOutOfSpace(ctrl.cfg) {
```

Comment:
I think we should move this check into `storage.Put`, because we also do self-profiling for the `pyroscope server` itself where it invokes `storage.Put` directly and with this current implementation this check won't be performed for those ingestions.